api/src/main/java/io/strimzi/api/kafka/model/common/metrics/JmxPrometheusExporterMetrics.java

```diff
@@ -31,7 +30,6 @@ public class JmxPrometheusExporterMetrics extends MetricsConfig {
     private ExternalConfigurationReference valueFrom;
 
     @Description("ConfigMap entry where the Prometheus JMX Exporter configuration is stored. ")
-    @JsonProperty(required = true)
```

Comment:
IIRC, you had to make it optional because of how the API is constructed. But it is required. So, we have to make sure:
* It is handled in the code
* Use the CEL validation (https://github.com/strimzi/strimzi-kafka-operator/issues/9417) to make this required that way (but this would need to be done right away). If you want, I can look at that when I get back to the office, but that would definitely delay this PR.

Reply:
FYI: I opened https://github.com/strimzi/strimzi-kafka-operator/pull/11068 for the CEL validation. Once merged, you would need to add the corresponding rule here.

Reply:
Addressed in this commit:
https://github.com/strimzi/strimzi-kafka-operator/pull/11051/commits/c04be8ac9bd35ec1523bd620631ab10f0edc5dd7